sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs

```diff
@@ -173,9 +178,15 @@ public BlobUriBuilder(Uri uri)
 
                 var startIndex = 0;
 
-                if (uri.IsHostIPEndPointStyle())
+                _isIPStyleUri = uri.IsHostIPEndPointStyle();
+                _isEmulatorStyleUri = (_isIPStyleUri || (
+                        !uri.Host.Contains(Constants.Blob.UriSubDomain) &&
```

Comment:
Is there a downside to restricting to the single account name defined [here ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Shared/Constants.cs#L111) and just calling this `isDevStoreUri`?

Reply:
Maybe?  I'm not very familiar with how the emulator works, since without custom hostname support it hasn't fit our workflows.  Restricting would remove some flexibility, if the emulator supports account names besides the dev store one (or if it were ever updated to do so).  

Mostly I didn't see upside to special-casing it.  If there is upside, I'm good with changing it.

Reply:
For non-emulator custom host names, I think this would end up populating the AccountName as well:
```c#
var blobUriBuilder = new BlobUriBuilder(new Uri("https://www.mycustomname.com/container/blob"));
Console.WriteLine(blobUriBuilder.AccountName); // prints container, but should be null
Console.WriteLine(blobUriBuilder.BlobContainerName) // prints blob, but should be container
```

We need some way to differentiate between these two cases - emulator-based Uri and [custom uri](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-custom-domain-name?tabs=azure-portal#step-5-test-your-custom-domain).
It looks like this was done based on the [ports ](https://github.com/Azure/azure-storage-net/blob/4bd9460e0b9b007bdb34eaa1329de2257f7a34dc/Lib/Common/Core/Util/CommonUtility.cs#L240) in Track 1. I think we could add a `_isPathStyleUri` that is true if `_IpStyleUri = true` or if the `Uri.Port` is contained in the list from the link above (we would need to copy over this list).

Reply:
That's helpful context, thanks!  You're right, this will fail for custom uris.  I'll get an update to this PR hopefully today, probably following the approach you suggest.